api/src/test/java/org/openmrs/EncounterTest.java

```diff
@@ -954,6 +954,66 @@ public void removeOrder_shouldNotFailWhenRemovingNonExistentOrder() {
 		encounter.removeOrder(new Order(123));
 	}
 	
+	/**
+	 * @see Encounter#addCondition(Condition)
+	 */
+	@Test
+	public void addCondition_shouldSetEncounterAttribute() {
+		// setup
+		Encounter encounter = new Encounter();
+		Condition condition = new Condition();
+		
+		// replay
+		encounter.addCondition(condition);
+		
+		// verify
+		assertTrue(condition.getEncounter().equals(encounter));
+	}
+	
+	/**
+	 * @see Encounter#addCondition(Condition)
+	 */
+	@Test
+	public void addCondition_shouldNotFailWithNullCondition() {
+		// setup
+		Encounter encounter = new Encounter();
+		
+		// replay
+		encounter.addCondition(null);
+		
+		// verify
+		assertEquals(0, encounter.getConditions().size());
+	}
```

Comment:
>I would not silently ignore one passing a null parameter. In other words, i would remove the if check and assume that the condition is not null. If it turns out to be null, better for the caller to automatically get a NullPointerException than chewing it quietly.

@dkayiwa this was by design that the method `addCondition(Condition)` is null safe.

I was simply following the behaviours  portrayed by existing `addXxxx(Xxxx)` methods in the codebase. An example is [addObs(Observation)](https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/Encounter.java#L344-L391)

Reply:
I feel that the existing behaviour is not a good one to copy. How can one add a null object and the api just silently ignore it?

Reply:
I agree with you here. The API shouldn't have back doors. Nice catch!